dali/dfuXRefLib/dfuxreflib.cpp

```diff
@@ -3010,3 +3032,158 @@ IPropertyTree * RunProcess(XRefCmd cmd, unsigned nclusters,const char **clusters
+        CPPUNIT_ASSERT_EQUAL_MESSAGE("Part number is incorrect",(unsigned)1,num);
+        CPPUNIT_ASSERT_EQUAL_MESSAGE("Max part number is incorrect",(unsigned)3,max);
+        CPPUNIT_ASSERT_EQUAL_MESSAGE("Stripe number is incorrect",(unsigned)0,stripeNum);
+        CPPUNIT_ASSERT_EQUAL_MESSAGE("Dir-per-part number is incorrect",(unsigned)0,dirPerPart);
+        CPPUNIT_ASSERT_EQUAL_MESSAGE("Replicate is incorrect",true,replicate);
+
+        // Standard file name with multiple parts is striped across directories, storage plane has numDevices set to 111
+        parseFileName("/var/lib/HPCCSystems/hpcc-data-two/d1/test/myname._10_of_30", mname.clear(), num, max, stripeNum, dirPerPart, replicate);
+        CPPUNIT_ASSERT_EQUAL_STR("/var/lib/HPCCSystems/hpcc-data-two/test/myname._$P$_of_30",mname.str());
+        CPPUNIT_ASSERT_EQUAL_MESSAGE("Part number is incorrect",(unsigned)10,num);
+        CPPUNIT_ASSERT_EQUAL_MESSAGE("Max part number is incorrect",(unsigned)30,max);
+        CPPUNIT_ASSERT_EQUAL_MESSAGE("Stripe number is incorrect",(unsigned)1,stripeNum);
+        CPPUNIT_ASSERT_EQUAL_MESSAGE("Dir-per-part number is incorrect",(unsigned)0,dirPerPart);
+        CPPUNIT_ASSERT_EQUAL_MESSAGE("Replicate is incorrect",true,replicate);
+
+        // Standard file name with multiple parts is striped across directories and each part has its own directory, storage plane has numDevices set to 111
+        parseFileName("/var/lib/HPCCSystems/hpcc-data-two/d1/test/42/myname._42_of_100", mname.clear(), num, max, stripeNum, dirPerPart, replicate);
+        CPPUNIT_ASSERT_EQUAL_STR("/var/lib/HPCCSystems/hpcc-data-two/test/myname._$P$_of_100",mname.str());
+        CPPUNIT_ASSERT_EQUAL_MESSAGE("Part number is incorrect",(unsigned)42,num);
+        CPPUNIT_ASSERT_EQUAL_MESSAGE("Max part number is incorrect",(unsigned)100,max);
+        CPPUNIT_ASSERT_EQUAL_MESSAGE("Stripe number is incorrect",(unsigned)1,stripeNum);
+        CPPUNIT_ASSERT_EQUAL_MESSAGE("Dir-per-part number is incorrect",(unsigned)42,dirPerPart);
+        CPPUNIT_ASSERT_EQUAL_MESSAGE("Replicate is incorrect",true,replicate);
+
+        // Test a longer part number
+        parseFileName("/var/lib/HPCCSystems/hpcc-data-two/d110/test/12345/myname._12345_of_100000", mname.clear(), num, max, stripeNum, dirPerPart, replicate);
+        CPPUNIT_ASSERT_EQUAL_STR("/var/lib/HPCCSystems/hpcc-data-two/test/myname._$P$_of_100000",mname.str());
+        CPPUNIT_ASSERT_EQUAL_MESSAGE("Part number is incorrect",(unsigned)12345,num);
+        CPPUNIT_ASSERT_EQUAL_MESSAGE("Max part number is incorrect",(unsigned)100000,max);
+        CPPUNIT_ASSERT_EQUAL_MESSAGE("Stripe number is incorrect",(unsigned)110,stripeNum);
+        CPPUNIT_ASSERT_EQUAL_MESSAGE("Dir-per-part number is incorrect",(unsigned)12345,dirPerPart);
+        CPPUNIT_ASSERT_EQUAL_MESSAGE("Replicate is incorrect",true,replicate);
+
+        // A file without a storage plane throws an exception
+        try {
```

Comment:
trivial/style: BSD style where open/close blocks are on a newline is the preferred style (+ new code).